### PR TITLE
refactor: fix and enhance the about menu on Linux

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: src/menu.rs [hunk 1]
```diff
@@ -187,6 +187,25 @@ impl Default for MenuBar {
   }
 }
 
+/// Application metadata for the [`MenuItem::About`] action.
+#[derive(Debug, Clone, Default)]
+pub struct AboutApplicationData {
```
Comment (amrbashir) [inline src/menu.rs]: Does `AboutMetadata` sound like a better name?

File: src/menu.rs [hunk 2]
```diff
@@ -187,6 +187,25 @@ impl Default for MenuBar {
   }
 }
 
+/// Application metadata for the [`MenuItem::About`] action.
```
Comment (amrbashir) [inline src/menu.rs:190]: We should document platform specific notes here instead of each field.

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/menu.rs (hunk 2 above)
Comment (lucasfernog) [reply src/menu.rs:190]: good idea, pushed

File: src/menu.rs (hunk 1 above)
Comment (lucasfernog) [reply src/menu.rs]: yeah, pushed

**Review by amrbashir — APPROVED** [review]

**Pull request merged**